cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -358,13 +344,22 @@ impl YaccParser {
         }
     }
 
-    fn parse_programs(&mut self, i: usize) -> YaccResult<usize> {
-        let mut j = i + 2; // Skip %%
-        j = self.parse_ws(j)?;
-        let prog = self.src[j..].to_string();
-        j = j + prog.len();
-        self.ast.add_programs(prog);
-        Ok(j)
+    fn parse_programs(&mut self, mut i: usize) -> YaccResult<usize> {
+        if let Some(j) = self.lookahead_is("%%", i) {
+            i = self.parse_ws(j)?;
+            if i == self.src.len() {
+                Ok(i)
+            }
```

Comment:
Join these two lines?

Reply:
Not sure I can, I still need `i` further down, so I can't just do
```
            if self.parse_ws(j)? == self.src.len() {
                Ok(i)
            }
```

Reply:
Ah sorry, I realised you were referring to the `} else` bit. Github showed me the wrong source code extract which mislead me.

Reply:
OK :)